runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingModeExecutionContext.java

```diff
@@ -552,7 +552,11 @@ public void flushState() {
       if (!cachedFiredTimers.hasNext()) {
         return null;
       }
-      return cachedFiredTimers.next();
+      TimerData nextTimer = cachedFiredTimers.next();
+      // system timers ( GC timer) must be explicitly deleted when delivered, to release the implied
+      // hold.
+      systemTimerInternals.deleteTimer(nextTimer);
```

Comment:
I think we should only do this if there is a hold to delete.

Reply:
@reuvenlax done